bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,104 @@
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+from pytz import country_timezones, timezone
+from datetime import datetime
+from discord.ext.commands import Cog, command
+from requests import get
+from discord import Embed
+from string import capwords
+
+
+class WeatherCog(Cog, name="Weather", description="get current weather information from a city"):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    @command(name='weather', help='Show weather information in your city', usage="{city}")
+    async def weather(self, ctx, *city_input):
+        city = capwords(" ".join(city_input))
+
+        # initialize Nominatim API
+        geolocator = Nominatim(user_agent="geoapiExercises")
```

Comment:
As I said in discord, all the stuff about getting the date for the city could also be put inside a function.

Reply:
I'm going to work on that during the week (tomorrow)